device/tt_silicon_driver.cpp

```diff
@@ -645,6 +659,19 @@ tt_SiliconDevice::tt_SiliconDevice(const std::string &sdesc_path, const std::str
     }
 }
 
+tt_SiliconDevice::tt_SiliconDevice(const std::string& sdesc_path, const std::string &ndesc_path, const std::set<chip_id_t> &target_devices, 
+                                   const uint32_t &num_host_mem_ch_per_mmio_device, const bool skip_driver_allocs,
+                                   const bool clean_system_resources, bool perform_harvesting, std::unordered_map<chip_id_t, uint32_t> simulated_harvesting_masks) : tt_device() {
+    construct_tt_silicon_device(sdesc_path, ndesc_path, target_devices, num_host_mem_ch_per_mmio_device, skip_driver_allocs, clean_system_resources, perform_harvesting, simulated_harvesting_masks);
+}
+
+tt_SiliconDevice::tt_SiliconDevice(tt::ARCH arch, const std::string &ndesc_path, const std::set<chip_id_t> &target_devices, 
+                                   const uint32_t &num_host_mem_ch_per_mmio_device, const bool skip_driver_allocs,
+                                   const bool clean_system_resources, bool perform_harvesting, std::unordered_map<chip_id_t, uint32_t> simulated_harvesting_masks) : tt_device() {
+    std::string sdesc_path = get_sdesc_path(arch);
```

Comment:
arch should not be passed, but inferred from target_devices, right?

Reply:
It would be ideal that `arch` isn't passed in.  An edge case is where the system has different cards in it (e.g. a WH and a GS) although it is reasonable if we don't want to support that case.

Something this eliminates is the ability of the caller to provide a custom SOC descriptor, i.e. a 1x1 or 2x2 grid.

Reply:
If a system has different cards, you reference which device_id (as reported by tt_ClusterDescriptor) you want to open, and you still don't pass arch or soc_desc